src/unitxt/artifact.py

```diff
@@ -138,6 +139,14 @@ class Artifact(Dataclass):
     )
     __id__: str = InternalField(default=None, required=False, also_positional=False)
 
+    __is_deprecated__: bool = NonPositionalField(
```

Comment:
We should add description of the features to the 

https://www.unitxt.ai/en/latest/docs/saving_and_loading_from_catalog.html

Reply:
@yoavkatz , I removed field `__is_deprecated__`  altogether, since we already have the `@deprecation` decorator from `deprecation_utils`.  Decorating class `RenameFields`, for example.  
And moved the link to the other Artifact to class `ArtifactLink` as you suggested.

Reply:
Thanks Dafna.  
 There is a difference between deprecating a class and deprecating an instance.  
For example, I can have a card saved as "card.sst" and at some point I want to call it "card.sst2".
I want to make the change be backward compatible, so "cards.sst" becomes a deprecated with warning.  It is also replaced with ArtficactLink  to "card.sst2".

Note that "Card" class was never deprecated

I also think a link should always be to acatalog entry and not an object.